TextReplacementHandler.apxc

```diff
@@ -0,0 +1,1771 @@
+     **/ 
+    public string populateMergeFields(String body, String openingDelimiter, String closingDelimiter) {   
+        System.debug('TextReplacementHandler::: populateMergeFields:: body: ' + body); 
+        if (body != null && body != ''){
+            Integer opening = body.countMatches(openingDelimiter);
+            Integer closing = body.countMatches(closingDelimiter);
+            Integer openingLen = openingDelimiter.length(); 
+            Integer closingLen = closingDelimiter.length(); 
+            System.debug(opening + ' -- ' + closing);
+        
+            //Go through each merge field to dynamically populate with actual info
+            for (Integer i = 0; i < opening; i++) {
+                Integer x = body.indexOf(openingDelimiter);
+                System.debug(x);
+                //Maximum of x unique merge fields, may have redundancies so exit if it has found all merge fields
+                if (x == -1) return body; 
+    
+                //Get the merge field
+                Integer y = body.indexOf(closingDelimiter);
+                System.debug(y);
+                String input = body.mid(x, y-x+closingLen);
+                System.debug(input);
+                String mergeField = body.mid(x+openingLen, y-x-openingLen);
+                System.debug(mergeField);
+                String obj = mergeField.substringBefore('.');
+                String field = mergeField.substringAfter('.');
+                System.debug('Object: ' + obj + ' -- Field: ' + field);
+    
+                //Find the replacement value
+                String output;
+                String val;
+                if (obj == 'Account' && theAccount != null) { 
+                    val = String.valueOf(theAccount.get(String.valueOf(field)));
+                }
+                else if (obj == 'User' && theUser != null) { 
+                    val = (String)theUser.get(String.valueOf(field));
+                }
+    
+    
+                //If the field doesn't have a value, leave as what the oldBody has
+                if (val != null) output = String.valueOf(val);
+                else output = '<<' + String.valueOf(mergeField) + '>>';
+                System.debug('Input: ' + input + ' -- Output: ' + output);
+                body = body.replace(input, output);
+            }
+            //Return the body with merge fields populated 
+            System.debug('TextReplacementHandler::: populateMergeFields:: replaced body: ' + body); 
+            return body; 
+        }
+        return body; 
+    } //public string populateMergeFields(String body, String openingDelimiter, String closingDelimiter)
+
+    
+    
+    /** Figure out the standard object type based on the recordId (based off of the first three characters)
+    ** Input: record Id that you want to know the object of 
+    ** Output: name of the object type of the record 
+    **/ 
+    public static String getObjectName(Id recordId) {
```

Comment:
I was going to suggest that you change this to use reflection (e.g. Schema Describes or EntityDefinition queries - I think it has a KeyPrefix column) so we can avoid all the hardcoded references, but also make sure it'll dynamically support new object types as Salesforce releases them.